plugins/jobtype/src/azkaban/jobtype/HadoopJobUtils.java

```
@@ -0,0 +1,409 @@
+    Set<String> applicationIds = new HashSet<String>();
+    Pattern p = Pattern.compile(".* (application_\\d+_\\d+).*");
+  
+    try {
+      br = new BufferedReader(new FileReader(logFile));
+      String input;
+  
+      // finds all the application IDs
+      while ((input = br.readLine()) != null) {
+        Matcher m = p.matcher(input);
+        if (m.find()) {
+          String appId = m.group(1);
+          if (!applicationIds.contains(appId)) {
+            applicationIds.add(appId);
+          }
+        }
+      }
+    } catch (IOException e) {
+      e.printStackTrace();
+      log.info("Error while trying to find applicationId for Spark log", e);
+    } finally {
+      try {
+        if (br != null)
+          br.close();
+      } catch (Exception e) {
+        // do nothing
+      }
+    }
+    return applicationIds;
+  }
+
+  /**
+   * <pre>
+   * Uses YarnClient to kill the job on HDFS.
+   * Using JobClient only works partially:
+   *   If yarn container has started but spark job haven't, it will kill
+   *   If spark job has started, the cancel will hang until the spark job is complete
+   *   If the spark job is complete, it will return immediately, with a job not found on job tracker
+   * </pre>
+   * 
+   * @param applicationId
+   */
+  public static void killJobOnCluster(String applicationId, Logger log) {
+    try {
+      YarnConfiguration yarnConf = new YarnConfiguration();
+      YarnClient yarnClient = YarnClient.createYarnClient();
+      yarnClient.init(yarnConf);
+      yarnClient.start();
+
+      String[] split = applicationId.split("_");
+      ApplicationId aid = ApplicationId.newInstance(Long.parseLong(split[1]),
+              Integer.parseInt(split[2]));
+
+      log.info("start klling application: " + aid);
+      yarnClient.killApplication(aid);
+      log.info("successfully killed application: " + aid);
+    } catch (Exception e) {
+      log.error("Failure while try to kill the hadoop job.  Will skip and continue", e);
```

How would the caller know if the given job was successfully killed? Should we throw an exception instead?